apps/emqx/src/emqx_channel.erl

```diff
@@ -2594,7 +2594,8 @@ do_check_sub_authzs2(ClientInfo, [TopicFilter = {Topic, _SubOpts} | More], Acc)
             emqx_topic:get_shared_real_topic(Topic)
         )
     of
-        %% TODO: support maximum QoS granted
+        %% TODO: support maximum QoS granted check by authz
```

Comment:
acl.conf already supports qos check, do you mean allow authz check to return a new QoS ?

Reply:
yes.
authz check only return `allow` or `deny`.
there is no support like this: ***authz backend denied qos2 but allowed and granted lower qos1***